### PR TITLE
Sort pattern aggregations fix postgres

### PR DESCRIPTION
**Pull request by srameshr**

### CODE REVIEW

**Comment by srameshr** [comment]
@dplewis @davimacedo

**Comment by srameshr** [comment]
@dplewis @davimacedo
Can you please review this PR?

**Comment by srameshr** [comment]
@dplewis @davimacedo 
Can you look into this? There is an error that is thrown during test cases init. Works fine locally
```
0.97s$ nvm install $NODE_VERSION
before_install.2
0.43s$ nvm use $NODE_VERSION
before_install.3
1.03s$ npm install -g greenkeeper-lockfile@1
before_install.4
0.01s$ sudo sed -i 's/port = 5433/port = 5432/' /etc/postgresql/11/main/postgresql.conf
before_install.5
0.01s$ sudo cp /etc/postgresql/{10,11}/main/pg_hba.conf
before_install.6
0.16s$ sudo service postgresql stop
before_install.7
2.54s$ sudo service postgresql start 11
install.npm
18.66s$ npm ci 
before_script.1
1.30s$ node -e 'require("./lib/index.js")'
before_script.2
0.22s$ psql -c 'create database parse_server_postgres_adapter_test_database;' -U postgres
0.17s$ psql -c 'CREATE EXTENSION postgis;' -U postgres -d parse_server_postgres_adapter_test_database
ERROR:  could not access file "$libdir/postgis-2.4": No such file or directory
The command "psql -c 'CREATE EXTENSION postgis;' -U postgres -d parse_server_postgres_adapter_test_database" failed and exited with 1 during .
```

**Comment by cbaker6** [comment]
@srameshr the error is will hopefully be fixed with #6505

**Comment by srameshr** [comment]
@cbaker6 That  PR seems to be open since the last two days. Please notify me when its merged.

**Comment by srameshr** [comment]
@cbaker6 Any updates on this?

**Comment by cbaker6** [comment]
@srameshr the main branch has the fix. Let me know if you see the Postgres error in the rest of your tests

**Comment by srameshr** [comment]
@cbaker6 Yes, I still get this error. I synced my branch with main master branch.

**Comment by cbaker6** [comment]
@srameshr from what I see it says your branch is 14 commits behind the master and I can't see that you ran the test again.

**Comment by srameshr** [comment]
@cbaker6 @dplewis @davimacedo There are some test cases failing which is totally unrelated to this commit. Can someone take a look at that?

**Comment by cbaker6** [comment]
@srameshr can run the test a couple of more times by committing? I would like to see if the same errors you just got are reoccurring

**Comment by srameshr** [comment]
@cbaker6 
I tried pushing it. It's not triggering the build again. Can you run it manually?

**Comment by dplewis** [comment]
@srameshr I just restarted the build. I'll keep an eye on it

**Comment by srameshr** [comment]
@dplewis @cbaker6 
It fails again.
```Cloud Code cloud jobs should run with master key Message: Unhandled promise rejection: ParseError: 101 Object not found.  
 1) PushController properly creates _PushStatus Message: Expected 'pending' to be 'running'.
```

**Comment by dplewis** [comment]
Thats a flaky test, nothing to do with this PR. It's really difficult to replicate locally otherwise it would be fixed.

File: spec/ParseQuery.Aggregate.spec.js [hunk 1]
```diff
@@ -225,6 +225,32 @@ describe('Parse.Query Aggregate testing', () => {
...
+        },
+      },
+    ];
+    Parse.Object.saveAll([obj1, obj2, obj3])
+      .then(() => {
+        const query = new Parse.Query(TestObject);
+        return query.aggregate(pipeline);
+      })
+      .then(results => {
+        expect(results.length).toEqual(5);
```
[the remark this answers is not in the document]
Comment (srameshr) [reply spec/ParseQuery.Aggregate.spec.js]: Something is messed up in my repo. Il create a new PR

**Comment by srameshr** [comment]
@dplewis 
https://github.com/parse-community/parse-server/pull/6522

**Comment by dplewis** [comment]
Closing via https://github.com/parse-community/parse-server/pull/6522

**Pull request closed without merging**